cmd/mongodb-healthcheck/main.go

```diff
@@ -42,108 +42,34 @@ func main() {
 	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGTERM, os.Interrupt)
 	defer stop()
 
-	app := tool.New("Performs health and readiness checks for MongoDB", GitCommit, GitBranch)
-
-	k8sCmd := app.Command("k8s", "Performs liveness check for MongoDB on Kubernetes")
-	livenessCmd := k8sCmd.Command("liveness", "Run a liveness check of MongoDB").Default()
-	_ = k8sCmd.Command("readiness", "Run a readiness check of MongoDB")
-	startupDelaySeconds := livenessCmd.Flag("startupDelaySeconds", "").Default("7200").Uint64()
-	component := k8sCmd.Flag("component", "").Default("mongod").String()
-
-	opts := zap.Options{
-		Encoder: getLogEncoder(),
-		Level:   getLogLevel(),
-	}
-	log := zap.New(zap.UseFlagOptions(&opts))
-	logf.SetLogger(log)
-
-	restoreInProgress, err := fileExists("/opt/percona/restore-in-progress")
-	if err != nil {
-		log.Error(err, "check if restore in progress")
-		os.Exit(1)
-	}
-
-	if restoreInProgress {
-		os.Exit(0)
-	}
-
-	sleepForever, err := fileExists("/data/db/sleep-forever")
-	if err != nil {
-		log.Error(err, "check if sleep-forever file exists")
-		os.Exit(1)
-	}
-	if sleepForever {
-		os.Exit(0)
-	}
-
-	cnf, err := db.NewConfig(
-		app,
-		pkg.EnvMongoDBClusterMonitorUser,
-		pkg.EnvMongoDBClusterMonitorPassword,
-	)
-
-	if err != nil {
-		log.Error(err, "new cfg")
-		os.Exit(1)
+	logPath := filepath.Join(psmdb.MongodDataVolClaimName, "logs", "mongodb-healthcheck.log")
+	logRotateWriter := lumberjack.Logger{
+		Filename:   logPath,
+		MaxSize:    100,
+		MaxAge:     1,
+		MaxBackups: 0,
+		Compress:   true,
 	}
-
-	command, err := app.Parse(os.Args[1:])
-	if err != nil {
-		log.Error(err, "Cannot parse command line")
-		os.Exit(1)
+	logOpts := zap.Options{
+		Encoder:    getLogEncoder(),
+		Level:      zapcore.DebugLevel,
```

Comment:
Could we set this through some env var? What if a user doesn't want this level of verbosity?

I think this should be configurable. Maybe we don't need to have it now, but maybe a task to have it. Log levels generally should be configurable, especially if we set it to a more verbose level and you can't change it.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1587/commits/c9efc0c4c68626bede2793fbd976b8f1b317688e